crypto/s2n_rsa_pss.c

```diff
@@ -176,14 +176,20 @@ static int s2n_rsa_pss_keys_match(const struct s2n_pkey *pub, const struct s2n_p
 
 static int s2n_rsa_pss_key_free(struct s2n_pkey *pkey)
 {
-    /* This object does not own the reference to the key --
-     * s2n_pkey handles it. */
+    POSIX_ENSURE_REF(pkey);
+    POSIX_ENSURE_REF(&pkey->key.rsa_key);
+    struct s2n_rsa_key *rsa_key = &pkey->key.rsa_key;
+    if (rsa_key->rsa == NULL) { return 0; }
+
```

Comment:
Nit: We avoid single line ifs in s2n-tls. Also, return S2N_SUCCESS to be clearer.
```suggestion
    if (rsa_key->rsa == NULL) {
        return S2N_SUCCESS;
    }

```